app/frontend/src/features/auth/jail/TOSSection.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { useCallback, useState } from "react";
+import Button from "../../../components/Button";
+import TextBody from "../../../components/TextBody";
+import TOS from "../../../components/TOS";
+import { acceptTOS } from "../../../service/jail";
+
+export default function TOSSection({
+  updateJailed,
+}: {
+  updateJailed: () => Promise<any>;
+}) {
+  const [completed, setCompleted] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  const accept = useCallback(async () => {
+    setLoading(true);
+    const info = await acceptTOS();
+    if (!info.isJailed) await updateJailed();
+    setLoading(false);
+    setCompleted(true);
+  }, [updateJailed]);
```

Comment:
```suggestion
  const accept = async () => {
    setLoading(true);
    const info = await acceptTOS();
    if (!info.isJailed) await updateJailed();
    setLoading(false);
    setCompleted(true);
  };
```
I don't think `useCallback` is necessary here as `updateJailed` is already stable from the `useCallback` coming from the parent component, since `dispatch` from Redux is going to be stable. Even if it's not stable, same comment as what I said in `Jail` - I think it's better to do this optimisation only when there's measured results that show this is slowing down the app. Otherwise we're trading increased memory usage for nothing.